### PR TITLE
Honor next URL when logged out

### LINKED ISSUE #2636: Does not redirect to desired page when logged out (next query isn't honoured)
## How to reproduce

1. Log out
2. Open a specific PH link (e.g. https://<PH instance>/person/338)
3. Login

## Expected behavior

I expect it to take me to the page I was intending to go to when I was logged out and not "Insights" page.

## Actual behavior

It takes me to the https://<PH instance>/insights...

## Environment

- PostHog cloud or self-managed? self-hosted
- PostHog version/commit: 1.17.0

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Make sure we redirect appropriately based on the /next url if you try to access a page and are not logged in.

Closes #2636

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Review by paolodamico — COMMENTED** [review]
Think it's worth adding some tests here, should be fairly straightforward and can help handle edge cases such as the encoding of the return URL. Looks good otherwise.

File: posthog/urls.py [hunk 1]
```diff
@@ -54,13 +54,24 @@ def login_view(request):
         email = request.POST["email"]
         password = request.POST["password"]
         user = cast(Optional[User], authenticate(request, email=email, password=password))
+        next_url = request.GET.get("next", "")
```
Comment (paolodamico) [inline posthog/urls.py:57]: Nit: You can just do `next_url = request.GET.get("next")`, if the param is not set, it'll default to `None`.

**Comment by yakkomajuri** [comment]
Cheers @paolodamico ! Test added and nit addressed

**Comment by yakkomajuri** [comment]
Ah yeah sorry I was with my head on the release and other things forgot to add the URL encoding tests

**Comment by paolodamico** [comment]
Oh no worries at all, team work haha!

**Pull request merged**